### PR TITLE
Tyler StatTracker refactor

### PR DESCRIPTION
**Pull request by tylerpporter**
Refactors majority of the methods on StatTracker. Breaks out smaller chunks of functionality and puts them in other classes.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -26,72 +22,43 @@ def initialize(games_data, teams_data, game_teams_data)
...
-    # Push all this logic to class method in Game & call that here?
-    @games_data.reduce({}) do |games_count, game|
-      if games_count[game.season.to_s].nil?
-        games_count[game.season.to_s] = 1
-      else
-        games_count[game.season.to_s] += 1
-      end
-      games_count
-    end
+    Game.count_of_games_by_season
```
Comment (tylerpporter) [inline lib/stat_tracker.rb:49]: The logic for this method now lives as a class method on the Games class. I refactored there as well, reducing the lines of code to 2.

File: lib/dependents.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require_relative 'game.rb'
+require_relative 'team.rb'
+require_relative 'game_teams.rb'
+require_relative 'hashable.rb'
+require_relative 'calculable'
```
Comment (reid-andrew) [inline lib/dependents.rb:5]: Let's have a conversation for a future refactor about whether we should include extensions on file names in require statements. Don't think I have strong feelings either way, just think we should be consistent across classes.

File: lib/game_teams.rb [hunk 3]
```diff
@@ -20,7 +22,7 @@ def self.win_percentage(grouped = @@all_game_teams_data.group_by(&:team_id))
     grouped = grouped
     grouped.map {|team, data| grouped[team] = data.map(&:result)}
     grouped.map do |team, data|
-      grouped[team] = (data.count("WIN")/data.size.to_f).round(2)
+      grouped[team] = (data.count("WIN")/data.size.to_f).round(5)
```
Comment (reid-andrew) [inline lib/game_teams.rb]: I don't think I'm immediately understanding why we changed the rounding & whether that has any impact anywhere else. Let's just talk through it so we're all on same page.

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -1,13 +1,9 @@
-require_relative 'game.rb'
-require_relative 'team.rb'
-require_relative 'game_teams.rb'
-require_relative 'hashable.rb'
-require_relative 'calculable'
+require_relative 'dependents'
```
Comment (reid-andrew) [inline lib/stat_tracker.rb:1]: Oh, cool. Don't think this even entered my mind as a possibility. Good catch.

**Pull request merged**